packaging/AppImage-builder/create_appimage.py

```diff
@@ -0,0 +1,102 @@
+# Copyright (c) 2023 UltiMaker
+# Cura is released under the terms of the LGPLv3 or higher.
+
+import argparse
+import os
+import shutil
+import subprocess
+
+from pathlib import Path
+
+from jinja2 import Template
+
+
+def prepare_workspace(dist_path, appimage_filename):
+    """
+    Prepare the workspace for building the AppImage.
+    :param dist_path: Path to the distribution of Cura created with pyinstaller.
+    :param appimage_filename: name of the AppImage file.
+    :return:
+    """
+    if not os.path.exists(dist_path):
+        raise RuntimeError(f"The dist_path {dist_path} does not exist.")
+
+    if os.path.exists(os.path.join(dist_path, appimage_filename)):
+        os.remove(os.path.join(dist_path, appimage_filename))
+
+    if not os.path.exists("AppDir"):
+        shutil.move(dist_path, "AppDir")
+    else:
+        print(f"AppDir already exists, assuming it is already prepared.")
+
+    copy_files("AppDir")
+
+
+def build_appimage(dist_path, version, appimage_filename):
+    """
+    Creates an AppImage file from the build artefacts created so far.
+    """
+    generate_appimage_builder_config(dist_path, version, appimage_filename)
+    create_appimage()
+    sign_appimage(dist_path, appimage_filename)
+
+
+def generate_appimage_builder_config(dist_path, version, appimage_filename):
+    with open(os.path.join(Path(__file__).parent, "AppImageBuilder.yml.jinja"), "r") as appimage_builder_file:
+        appimage_builder = appimage_builder_file.read()
+
+    template = Template(appimage_builder)
+    appimage_builder = template.render(app_dir = "./AppDir",
+                                       icon = "cura-icon.png",
+                                       version = version,
+                                       arch = "x86_64",
+                                       file_name = appimage_filename)
+
+    with open(os.path.join(Path(__file__).parent, "AppImageBuilder.yml"), "w") as appimage_builder_file:
+        appimage_builder_file.write(appimage_builder)
+
+
+def copy_files(dist_path):
+    """
+    Copy metadata files for the metadata of the AppImage.
+    """
+    copied_files = {
+        os.path.join("..", "icons", "cura-icon.svg"): os.path.join("usr", "share", "icons", "hicolor", "scalable", "apps", "cura-icon.svg"),
+        os.path.join("..", "icons", "cura-icon_64x64.png"): os.path.join("usr", "share", "icons", "hicolor", "64x64", "apps", "cura-icon.png"),
+        os.path.join("..", "icons", "cura-icon_128x128.png"): os.path.join("usr", "share", "icons", "hicolor", "128x128", "apps", "cura-icon.png"),
+        os.path.join("..", "icons", "cura-icon_256x256.png"): os.path.join("usr", "share", "icons", "hicolor", "256x256", "apps", "cura-icon.png"),
+        os.path.join("..", "icons", "cura-icon_256x256.png"): "cura-icon.png",
+    }
+
+    # TODO: openssl.cnf ???
```

Comment:
Does this need to happen?

Reply:
Need to double check, if you can overload the openssl config. But we weren't doing that before, since that file doesn't exist.
I think it only be an issue on Windows